orcoursetrion/lib/github.py

```diff
@@ -107,6 +141,119 @@ def create_repo(self, org, repo, description):
+
+        Args:
+            org (str): Organization to create the repo in.
+            team_name (str): Name of team to create.
+            read_only (bool): If false, read/write, if true read_only.
+
+        Raises:
+            GitHubUnknownError
+            requests.RequestException
+        Returns:
+            dict: Team dictionary
+                  (https://developer.github.com/v3/orgs/teams/#response)
+        """
+        if read_only:
+            permission = 'pull'
+        else:
+            permission = 'push'
+
+        create_url = '{url}orgs/{org}/teams'.format(
+            url=self.api_url,
+            org=org
+        )
+        response = self.session.post(create_url, json={
+            'name': team_name,
+            'permission': permission
+        })
+        if response.status_code != 201:
+            raise GitHubUnknownError(response.text)
+        return response.json()
+
+    def put_team(self, org, team_name, read_only, members):
+        """Create a team in a github organization.
+
+        Utilize
+        https://developer.github.com/v3/orgs/teams/#list-teams,
+        https://developer.github.com/v3/orgs/teams/#create-team,
+        https://developer.github.com/v3/orgs/teams/#list-team-members,
+        https://developer.github.com/v3/orgs/teams/#add-team-membership,
+        and
+        https://developer.github.com/v3/orgs/teams/#remove-team-membership.
+        to create a team and/or replace an existing team's membership
+        with the ``members`` list.
+
+        Args:
+            org (str): Organization to create the repo in.
+            team_name (str): Name of team to create.
+            read_only (bool): If false, read/write, if true read_only.
+            members (list): List of github usernames to add to the
+                              team. If none, membership changes won't occur
+
+        Raises:
+            GitHubUnknownError
+            requests.RequestException
+
+        Returns:
+            dict: The team dictionary
+                (https://developer.github.com/v3/orgs/teams/#response-1)
+
+        """
+        try:
+            team_dict = self._find_team(org, team_name)
+        except GitHubNoTeamFound:
+            team_dict = self._create_team(org, team_name, read_only)
+
+        # Just get the team and exit if no members are given
+        if members is None:
+            return team_dict
+
+        # Have the team, now replace member list with the one we have
+        members_url = '{url}teams/{id}/members'.format(
+            url=self.api_url,
+            id=team_dict['id']
+        )
+        existing_members = self._get_all(members_url)
+        # Filter list of dicts down to just username list
+        existing_members = [x['login'] for x in existing_members]
+
+        # Grab everyone that should no longer be members
+        remove_members = dict(
+            [(x, False) for x in existing_members if x not in members]
+        )
```

Comment:
I believe that to be true, and want an uncaught exception if it ever proves otherwise. This is from https://developer.github.com/v3/orgs/teams/#response-5